pkg/gadgets/tcptop/tracer/tracer.go

```diff
@@ -0,0 +1,277 @@
+//go:build linux
+// +build linux
+
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this bio except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"time"
+	"unsafe"
+
+	containercollection "github.com/kinvolk/inspektor-gadget/pkg/container-collection"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/tcptop/types"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/link"
+)
+
+// #include <linux/types.h>
+// #include "./bpf/tcptop.h"
+// #include <arpa/inet.h>
+// #include <stdlib.h>
+//
+//char *string_ip(const void *addr, int ip_type) {
+//	socklen_t size;
+//	char *ip;
+//
+//  // Should not occur because eBPF code already filter on this.
+//	if (ip_type != AF_INET && ip_type != AF_INET6)
+//		return NULL;
+//
+//	size = sizeof *ip * INET6_ADDRSTRLEN;
+//	ip = malloc(size);
+//  if (ip == NULL)
+//		return NULL;
+//
+//	inet_ntop(ip_type, addr, ip, size);
+//
+//	return ip;
+//}
+//
+// char *dst_addr(struct ip_key_t *key) {
+// 	return string_ip(&key->daddr, key->family);
+// }
+//
+// char *src_addr(struct ip_key_t *key) {
+// 	return string_ip(&key->saddr, key->family);
+// }
```

Comment:
Remember to remove it before merging if it is not useful, otherwise please write a comment describing why it is commented.

Reply:
This seems strange at first glance but commenting C code is the way to [write C code (using Cgo) withing golang](https://matrix-org.github.io/go-neb/pkg/C/index.html#hdr-Go_references_to_C).
You can check I actually use `src_addr()` and `dst_addr()` somewhere in the code.